### PR TITLE
[demo][external-assets] create stand alone check for pipes

### PR DESCRIPTION
**Pull request by Ramshackle-Jamathon**
## Summary & Motivation

creates a separate image to be used in k8s and stand alone asset check for the pipes example.

currently failing with the following error

```
[pipes] did not receive any messages from external process. Check stdout / stderr logs from the external process if possible.
PipesEnvContextInjector: Attempted to inject context directly, typically as an environment variable.
PipesK8sPodLogsMessageReader: Attempted to read messages by extracting them from kubernetes pod logs directly.
```

### CODE REVIEW

File: examples/experimental/external_assets/external_assets/pipes/__init__.py [hunk 1]
```diff
@@ -23,13 +20,29 @@
...
+                }
+            ]
+        },
+    ).get_results()
+
+
+@asset_check(asset="telem_post_processing")
+def telem_post_processing_check(context, k8s_pipes_client: PipesK8sClient):
+    context.log.info(context)
+    values = yield from k8s_pipes_client.run(
```
Comment (schrockn) [inline examples/experimental/external_assets/external_assets/pipes/__init__.py:42]: You should be able to use @alangenfeld 's latest change to do this properly

File: examples/experimental/external_assets/external_assets/pipes/__init__.py [hunk 2]
```diff
@@ -23,13 +20,29 @@
...
-        image="pipes-dogfood:latest",
+        image="pipes-materialize:latest",
+        base_pod_spec={
+            "containers": [
+                {
+                    "imagePullPolicy": "Never",
+                }
+            ]
+        },
+    ).get_results()
```
Comment (schrockn) [inline examples/experimental/external_assets/external_assets/pipes/__init__.py:36]: Also recommend moving this to `get_materialize_result`

File: examples/experimental/external_assets/external_assets/pipes/__init__.py [hunk 3]
```diff
@@ -38,6 +51,8 @@ def telem_post_processing(context: AssetExecutionContext, k8s_pipes_client: Pipe
             ]
         },
     ).get_results()
+    context.log.info(values)
```
Comment (alangenfeld) [inline examples/experimental/external_assets/external_assets/pipes/__init__.py]: ^ `get_results()` -> `get_asset_check_result()`

**Comment by Ramshackle-Jamathon** [comment]
blocked by https://github.com/dagster-io/dagster/pull/17123 or some other solve for this

**Review by alangenfeld — APPROVED** [review]

File: examples/experimental/external_assets/external_assets/pipes/__init__.py [hunk 4]
```diff
@@ -23,21 +20,24 @@
...
+
+
+@asset_check(asset="telem_post_processing")
+def telem_post_processing_check(context, k8s_pipes_client: PipesK8sClient):
+    yield from k8s_pipes_client.run(
+        context=context,
+        namespace="default",
+        image="pipes-check:latest",
+        env={"DAGSTER_PIPES_IS_DAGSTER_PIPED_PROCESS": encode_env_var(True)},
+    ).get_materialize_result()
```
Comment (alangenfeld) [inline examples/experimental/external_assets/external_assets/pipes/__init__.py:40]: `.get_asset_check_result()`

**Pull request merged**